src/OrleansSQLUtils/Storage/DbConnectionFactory.cs

```diff
@@ -16,6 +16,13 @@ private static CachedFactory GetFactory(string invariantName)
             // Seeks for database provider factory classes from GAC or as indicated by
             // the configuration file, see at <see href="https://msdn.microsoft.com/en-us/library/dd0w4a2z%28v=vs.110%29.aspx">Obtaining a DbProviderFactory</see>.       
 
+            if (invariantName == AdoNetInvariants.InvariantNamePostgreSql)
```

Comment:
I thought eyebrows would be raised here.  :)
I can't say I'm an expert here, but removing this line prevents Npgsql from being found.  I think it works for SQL Server Client, b/c it's in the GAC.  Personally I don't see a problem with the code, b/c the project now "knows" about Npgsql via a dependency on the Npgsql nuget package.